library/core/src/slice/mod.rs

```diff
@@ -2884,9 +2885,19 @@ impl<T> [T] {
     /// This sort is unstable (i.e., may reorder equal elements), in-place (i.e., does not
     /// allocate), and *O*(*n* \* log(*n*)) worst-case.
     ///
-    /// If `T: Ord` does not implement a total order the resulting order is unspecified. All
-    /// original elements will remain in the slice and any possible modifications via interior
-    /// mutability are observed in the input. Same is true if `T: Ord` panics.
+    /// If the implementation of [`Ord`] for `T` does not implement a [total order] the resulting
+    /// order of elements in the slice is unspecified. All original elements will remain in the
+    /// slice and any possible modifications via interior mutability are observed in the input. Same
+    /// is true if the implementation of [`Ord`] for `T` panics.
+    ///
+    /// Sorting types that only implement [`PartialOrd`] such as [`f32`] and [`f64`] requires
+    /// additional precautions. For example Rust defines `NaN != NaN`, which doesn't fulfill the
```

Comment:
This is an IEEE754 requirement about the normal operator's behavior so there's no need to say "Rust defines".
```suggestion
    /// additional precautions. For example, `NaN != NaN`, which doesn't fulfill the
```

Reply:
Fair enough, I've changed it to `f32::NAN != f32::NAN` to be actual code, to make the connection to the previously named types more apparent.